test/powershell/Modules/Microsoft.PowerShell.Security/CmsMessage.Tests.ps1

```diff
@@ -0,0 +1,400 @@
+    [IO.File]::WriteAllBytes($certLocation, $certBytes)
+
+    return $certLocation
+}
+
+
+Describe "CmsMessage cmdlets and Get-PfxCertificate basic tests" -Tags "CI" {
+    
+    BeforeAll {
+        $certLocation = Create-TestCertificate
+        $certLocation | Should Not BeNullOrEmpty | Out-Null
+    }
+
+    It "Verify Get-PfxCertificate -FilePath" {
+        $cert = Get-PfxCertificate -FilePath $certLocation
+        $cert.Subject | Should Be "CN=MyDataEnciphermentCert"
+    }
+
+    It "Verify Get-PfxCertificate -LiteralPath" {
+        $cert = Get-PfxCertificate -LiteralPath $certLocation
+        $cert.Subject | Should Be "CN=MyDataEnciphermentCert"
+    }
+
+    It "Verify Get-PfxCertificate positional argument" {
+        $cert = Get-PfxCertificate $certLocation
+        $cert.Subject | Should Be "CN=MyDataEnciphermentCert"
+    }
+
+    It "Verify CMS message recipient resolution by path" -Skip:(!$IsWindows) {
+        $ers = $null
+        $recipient = [System.Management.Automation.CmsMessageRecipient] $certLocation
+        $recipient.Resolve($ExecutionContext.SessionState, "Encryption", [ref] $ers)
+
+        $recipient.Certificates[0].Subject | Should Match 'CN=MyDataEnciphermentCert'
+    }
+
+    It "Verify CMS message recipient resolution by cert" -Skip:(!$IsWindows) {
+        $ers = $null
+        $cert = Get-PfxCertificate $certLocation
+        $recipient = [System.Management.Automation.CmsMessageRecipient] $cert
+        $recipient.Resolve($ExecutionContext.SessionState, "Encryption", [ref] $ers)
+
+        $recipient.Certificates[0].Subject | Should Match 'CN=MyDataEnciphermentCert'
+    }
+
+    It "Verify a CMS message can be protected / unprotected" -Skip:(!$IsWindows) {
+        $protected = "Hello World","How are you?" | Protect-CmsMessage -To $certLocation
+        $protected.IndexOf("-----BEGIN CMS-----") | Should Be 0
+
+        $message = $protected | Get-CmsMessage
+        $message.Recipients[0].IssuerName | Should Be "CN=MyDataEnciphermentCert"
+
+        $decrypted = $message | Unprotect-CmsMessage -To $certLocation
+        $decrypted | Should Be "Hello World`r`nHow are you?"
+
+        $decrypted = $protected | Unprotect-CmsMessage -To $certLocation
+        $decrypted | Should Be "Hello World`r`nHow are you?"
+    }
+}
+
+
+Describe "CmsMessage cmdlets thorough tests" -Tags "Feature" {
+
+    BeforeAll {
+        if ($IsWindows)
+        {
+            $certLocation = Create-TestCertificate
+            $certLocation | Should Not BeNullOrEmpty | Out-Null
+
+            if ($IsCoreCLR)
+            {
+                # PKI module is not available for PowerShell Core, so we need to use Windows PowerShell to import the cert
+                $fullPowerShell = Join-Path "$env:SystemRoot" "System32\WindowsPowerShell\v1.0\powershell.exe"
+
+                try {
+                    $modulePathCopy = $env:PSMODULEPATH
+                    $env:PSMODULEPATH = $null
```

Comment:
It's a hack to import a certificate using PKI module from powershell core. PKI module is not carried by powershell core, but we only run those tests on windows platform, which makes this hack possible.